elements/howto-menu/howto-menu.js

```diff
@@ -0,0 +1,247 @@
+     */
+    _firstMenuItem() {
+      let child = this.firstElementChild;
+      while (child) {
+        if (this._isMenuItem(child)) {
+          return child;
+        }
+        child = next.nextElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Returns the last child which is a menuitem.
+     */
+    _lastMenuItem() {
+      let child = this.lastElementChild;
+      while (child) {
+        if (this._isMenuItem(child)) {
+          return child;
+        }
+        child = next.previousElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Checks if a node is a 'menuitem', 'menuitemcheckbox' or 'menuitemradio'.
+     */
+    _isMenuItem(node) {
+      return MENUITEM_ARIA_ROLES.includes(node.getAttribute('role'));
+    }
+
+    /**
+     * Returns a menu item coming after the one passed as an argument, or null.
+     */
+    _nextMenuItem(node) {
+      let next = node.nextElementSibling;
+      while (next) {
+        if (this._isMenuItem(node)) {
+          return next;
+        }
+        next = next.nextElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Returns a menu item coming before the one passed as an argument, or null.
+     */
+    _previousMenuItem(node) {
+      let prev = node.previousElementSibling;
+      while (prev) {
+        if (this._isMenuItem(node)) {
+          return prev;
+        }
+        prev = prev.previousElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Handles keyboard interaction.
+     */
+    _onKeyDown(event) {
+      let el = event.target;
+      switch (event.keyCode) {
+        case KEYCODE.DOWN:
+          // If arrow down, move to next item. Wrap if necessary.
+          event.preventDefault();
+          let next = this._nextMenuItem(el) || this._firstMenuItem();
+          next.focus();
+          break;
+       case KEYCODE.UP:
+         // If arrow up, move to previous item. Wrap if necessary.
+         event.preventDefault();
+         let prev = this._previousMenuItem(el) || this._lastMenuItem();
+         prev.focus();
+         break;
+       case KEYCODE.ESCAPE:
+         // If escape, close the menu.
+         event.preventDefault();
+         this.opened = false;
+         break;
+       case KEYCODE.TAB:
+         // If tab, close the menu.
+         event.preventDefault();
+         this.opened = false;
+         break;
+       default:
+         break;
+      }
+      // If letter key, move to an item which starts with that letter.
+      if (event.keyCode >= KEYCODE.A && event.keyCode <= KEYCODE.Z) {
+        for (let child of this.children) {
+          if (child.innerText.trim()[0] === event.key) {
+            event.preventDefault();
+            child.focus();
+            break;
+          }
+        }
+      }
+    }
+
+    open() {
+      this._addTabindex();
+      let first = this._firstMenuItem();
+      first && first.focus();
+    }
+
+    set opened(isOpened) {
+      if (!!isOpened) {
```

Comment:
small nit. would you mind replacing this with a call to `Boolean()`, similar to
https://github.com/GoogleChrome/howto-components/blob/master/elements/howto-checkbox/howto-checkbox.js#L125

My concern is not everyone understands what the !! trick does.